packages/core/src/rules/common/spec.ts

```diff
@@ -37,6 +37,18 @@ export const OasSpec: Oas3Rule | Oas2Rule = () => {
         }
       }
 
+      if (oasVersion !== 'oas2') {
```

Comment:
Better to add something like `oneOfProperty` into the type tree here https://github.com/Redocly/openapi-cli/blob/master/packages%2Fcore%2Fsrc%2Ftypes%2Foas3.ts#L121